hbase-rsgroup/src/test/java/org/apache/hadoop/hbase/rsgroup/TestRSGroupsAdmin2.java

```diff
@@ -459,4 +463,197 @@ public boolean evaluate() throws Exception {
     Assert.assertEquals(null, rsGroupAdmin.getRSGroupInfo(fooGroup.getName()));
   }
 
+  @Test
+  public void testFailedMoveWhenMoveServer() throws Exception {
+    final RSGroupInfo newGroup = addGroup(getGroupName(name.getMethodName()), 1);
```

Comment:
_addGroup_ already invokes _ moveServerRegionsFromGroup_ indirectly, and adds an RS to it. We could simplify this test if we just create a new group from _RSGroupAdminServer.addRSGroup_, pick one of the RSes from assignMap, change one of its regions state, then call later call _RSGroupAdminServer.moveServers_?

Reply:
Yes, good idea.

Reply:
In this case, I have used rsGroupAdmin.addRSGroup to add a new group without adding server. Then I create a multi-region table, choose a region to change state, and record the server that the region we changed on. In the last call RSGroupAdminServer.moveServers to move the recorded server, and check if  the error message contains the region name.